src/Hl7.Fhir.Base/FhirPath/Parser/Grammar.cs

```diff
@@ -65,47 +65,51 @@ private static IResult<P.Quantity> quantityParser(IInput i)
             .Named("BracketExpr");
 
         public static readonly Parser<Expression> EmptyList =
-            Parse.Char('{').Token().Then(c => Parse.Char('}').Token())
-                    .Select(v => NewNodeListInitExpression.Empty);
+            Parse.Char('{').TokenIgnoreComments().Then(c => Parse.Char('}').TokenIgnoreComments())
+                    .Select(v => NewNodeListInitExpression.Empty)
+                    .Positioned();
 
         public static Parser<Expression> Function(Expression context)
         {
             return
+                (
                 from n in Lexer.Identifier.Select(name => name)
-                from lparen in Parse.Char('(').Token()
-                from paramList in Parse.Ref(() => FunctionParameter(n).Named("parameter")).DelimitedBy(Parse.Char(',').Token()).Optional()
-                from rparen in Parse.Char(')').Token()
-                select new FunctionCallExpression(context, n, TypeSpecifier.Any, paramList.GetOrElse(Enumerable.Empty<Expression>()));
+                from lparen in Parse.Char('(').TokenIgnoreComments()
+                from paramList in Parse.Ref(() => FunctionParameter(n).Named("parameter")).DelimitedBy(Parse.Char(',').TokenIgnoreComments()).Optional()
+                from rparen in Parse.Char(')').TokenIgnoreComments()
+                select new FunctionCallExpression(context, n, TypeSpecifier.Any, paramList.GetOrElse(Enumerable.Empty<Expression>())))
+                .Positioned();
         }
 
         public static Parser<Expression> FunctionParameter(string name) =>
             // Make exception for is() and as() FUNCTIONS (operators are handled elsewhere), since they don't
             // take a normal parameter, but an identifier (which is not normally a FhirPath type)
-            name != "is" && name != "as" && name != "ofType" ? Grammar.Expression : TypeSpec.Select(s => new ConstantExpression(s));
+            (name != "is" && name != "as" && name != "ofType" ? Grammar.Expression : TypeSpec.Select(s => new IdentifierExpression(s))).Positioned();
 
 
         public static Parser<Expression> FunctionInvocation(Expression focus)
         {
             return Function(focus)
-                .Or(Lexer.Identifier.Select(i => new ChildExpression(focus, i)))
+                .Or(Lexer.Identifier.Select(i => new ChildExpression(focus, i)).Positioned())
                 //.XOr(Lexer.Axis.Select(a => new AxisExpression(a)))
-                .Token();
+                .TokenIgnoreComments();
         }
 
         public static readonly Parser<Expression> Term =
             Literal
             .Or(FunctionInvocation(AxisExpression.That))
-            .Or(Lexer.ExternalConstant.Select(n => BuildVariableRefExpression(n))) //Was .XOr(Lexer.ExternalConstant.Select(v => Eval.ExternalConstant(v)))
+            .XOr(Lexer.ExternalConstant.Select(n => BuildVariableRefExpression(n)).Positioned()) //Was .XOr(Lexer.ExternalConstant.Select(v => Eval.ExternalConstant(v)))
```

Comment:
@Kasdejong @ewoutkramer This is the line that I was referring to in Slack.

Reply:
I am testing this right now, no errors yet when removing it. I'm pretty sure this can be XOR again!

Reply:
It is now Xor and all tests are working, so I'll assume this is ok now.